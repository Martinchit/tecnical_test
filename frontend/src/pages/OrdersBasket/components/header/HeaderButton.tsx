import styled from 'styled-components';
import { Button } from 'react-bootstrap';

export const HeaderButton = styled(Button)`
  margin-left: 2%;
  width: 10%;

  @media (max-width: 767px) {
    width: 45%;
    margin-top: 3%;
    font-size: 0.8rem;
  }

  &:focus {
    outline: none;
    box-shadow: none;
  }
`;
