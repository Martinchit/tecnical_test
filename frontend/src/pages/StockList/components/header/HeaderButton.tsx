import styled from 'styled-components';
import { Button } from 'react-bootstrap';

export const HeaderButton = styled(Button)`
  margin-left: 1%;
  width: 10%;
  min-width: 85px;

  @media (max-width: 767px) {
    width: 50%;
    margin-top: 5%;
    font-size: 0.8rem;
  }

  &:focus {
    outline: none;
    box-shadow: none;
  }
`;
