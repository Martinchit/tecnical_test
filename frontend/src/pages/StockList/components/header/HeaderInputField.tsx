import styled from 'styled-components';
import { FormControl } from 'react-bootstrap';

export const HeaderInputField = styled(FormControl)`
  width: 30%;
  min-width: 250px;

  @media (max-width: 767px) {
    width: 100%;
  }
`;
