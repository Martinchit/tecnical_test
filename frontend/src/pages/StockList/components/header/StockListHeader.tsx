import React from 'react';
import { HeaderContainer } from './HeaderContainer';
import { HeaderInputField } from './HeaderInputField';
import { HeaderButton } from './HeaderButton';

interface StockListHeaderProps {
  onSearchValueChangeHandler: (value: string) => void;
  onSearchButtonClickHandler: () => void;
  onClearButtonClickHandler: () => void;
  searchValue: string;
  buttonsDisable: boolean;
  isStocksFiltered: boolean;
}

export const StockListHeader: React.FC<StockListHeaderProps> = ({
  onClearButtonClickHandler,
  onSearchValueChangeHandler,
  onSearchButtonClickHandler,
  searchValue,
  buttonsDisable,
  isStocksFiltered,
}) => {
  return (
    <HeaderContainer>
      <HeaderInputField
        value={searchValue}
        onChange={({ target }) => onSearchValueChangeHandler(target.value)}
        placeholder="Search Stock by Stock Code"
      />
      <HeaderButton
        disabled={buttonsDisable}
        variant="primary"
        onClick={onSearchButtonClickHandler}
      >
        Search
      </HeaderButton>
      <HeaderButton
        disabled={buttonsDisable || !isStocksFiltered}
        variant="warning"
        onClick={onClearButtonClickHandler}
      >
        Clear
      </HeaderButton>
    </HeaderContainer>
  );
};
