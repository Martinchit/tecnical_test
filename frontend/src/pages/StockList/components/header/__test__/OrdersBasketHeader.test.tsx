import React from 'react';
import { shallow } from 'enzyme';

import { StockListHeader } from '../StockListHeader';
import { HeaderButton } from '../HeaderButton';
import { HeaderInputField } from '../HeaderInputField';

describe('StockListHeader', () => {
  const props = {
    onClearButtonClickHandler: jest.fn(),
    onSearchValueChangeHandler: jest.fn(),
    onSearchButtonClickHandler: jest.fn(),
    searchValue: 'test',
    buttonsDisable: false,
    isStocksFiltered: false,
  };

  const restoreMock = () => {
    props.onClearButtonClickHandler.mockRestore();
    props.onSearchValueChangeHandler.mockRestore();
    props.onSearchButtonClickHandler.mockRestore();
  };

  it('renders correctly', () => {
    const wrapper = shallow(<StockListHeader {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('triggers onSearchButtonClickHandler', () => {
    const wrapper = shallow(<StockListHeader {...props} />);
    wrapper.find(HeaderButton).first().simulate('click');
    expect(props.onSearchButtonClickHandler).toHaveBeenCalled();
    restoreMock();
  });

  it('triggers onClearButtonClickHandler', () => {
    const wrapper = shallow(<StockListHeader {...props} />);
    wrapper.find(HeaderButton).at(1).simulate('click');
    expect(props.onClearButtonClickHandler).toHaveBeenCalled();
    restoreMock();
  });

  it('triggers onSearchValueChangeHandler', () => {
    const wrapper = shallow(<StockListHeader {...props} />);
    wrapper
      .find(HeaderInputField)
      .first()
      .simulate('change', { target: { value: 'test' } });
    expect(props.onSearchValueChangeHandler).toHaveBeenCalled();
    expect(props.onSearchValueChangeHandler).toHaveBeenCalledWith('test');
    restoreMock();
  });
});
