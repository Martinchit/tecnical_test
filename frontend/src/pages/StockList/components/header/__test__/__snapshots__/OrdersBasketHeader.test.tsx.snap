// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockListHeader renders correctly 1`] = `
<Styled(styled.div)>
  <Styled(FormControl)
    onChange={[Function]}
    placeholder="Search Stock by Stock Code"
    value="test"
  />
  <Styled(Button)
    disabled={false}
    onClick={[MockFunction]}
    variant="primary"
  >
    Search
  </Styled(Button)>
  <Styled(Button)
    disabled={true}
    onClick={[MockFunction]}
    variant="warning"
  >
    Clear
  </Styled(Button)>
</Styled(styled.div)>
`;
